x/qgb/abci_test.go

```diff
@@ -83,4 +145,36 @@ func TestValsetSetting(t *testing.T) {
 	require.Equal(t, uint64(1), pk.GetLatestAttestationNonce(ctx))
 }
 
-// Add data commitment window tests
+func TestSetDataCommitment(t *testing.T) {
```

Comment:
> why is the test named TestSetDataCommitment when the function under test is SetAttestationRequest?

An attestation request can be either a valset or a data commitment. that's why it is setting an attestation request.

Considering the refactor, I tried doing that and went for a table test. However, I kept having flaky tests because of the in-process network used. Thus, I separated the tests again.

Reply:
Thanks for sharing. This convo can be resolved and I'll defer to your judgement on whether we want an issue to investigate the flaky tests and refactor to a table test

Reply:
> I kept having flaky tests because of the in-process network used

do we know why the test was flaky? was it just because we need certain things to occur in order?

Reply:
> was it just because we need certain things to occur in order

:100: